### DISCUSSION TITLE
add hebrew

### DISCUSSION THREAD

**Discussion by maorisraelii** [xxyzz/WordDumb#194] [Ideas]
can you add Hebrew translation for English books?
I can help you if you need help

**Comment by xxyzz** [comment]
This would require English to Hebrew data extracted from the Hebrew Wiktionary or any other free dictionary.

You could take a look of the [wiktextract](https://github.com/tatuylonen/wiktextract), or try to parse these dump files: https://dumps.wikimedia.org/other/enterprise_html/runs/20240220/ , https://download.kiwix.org/zim/wiktionary/

**Reply by maorisraelii** [reply]
I download this "wiktionary_he_all_maxi_2024-02.zim" and now what i need to do?

**Reply by xxyzz** [reply]
You could check out https://github.com/SrGnis/wikjote, it processes the zim file.

And https://github.com/Vuizur/ruwiktionary-htmldump-parser processes the wikimedia html dump file.

**Reply by maorisraelii** [reply]
ok, I build 205K word english to hebrew PO what now?

**Reply by xxyzz** [reply]
Could you publish the data somewhere so it could be used by https://github.com/xxyzz/Proficiency to create files used by this plugin?

**Reply by maorisraelii** [reply]
[en_to_he.zip](https://github.com/xxyzz/WordDumb/files/14462537/en_to_he.zip)

not the best dict but i think its ok for start

**Reply by xxyzz** [reply]
Ideally the data would contain more data, like pos, full gloss, example sentences, ipa, and could update monthly with the new dump file like kaikki.org and dbnary project.

**Reply by maorisraelii** [reply]
I don't know how to make it update regularly, can u implement this for a start and also give me a PO from another language like (en to en) and I will translate from the other PO ?
thanks

**Reply by xxyzz** [reply]
PO files are for localization, not suitable for creating word wise file. You could take a look of the data created by [kaikki.org](https://kaikki.org) and [dbnary](https://kaiko.getalp.org/about-dbnary/) , and the code of [Proficiency](https://github.com/xxyzz/Proficiency). I don't think these data could be translated manually...

Maybe I misunderstand your intention? Are you trying to translate the plugin ui texts to Hebrew?

**Reply by maorisraelii** [reply]
I would like to add Hebrew to Wordwise, so that when I read an English book, I can enable the Hebrew Wordwise feature. However, I am not sure what I need to do in order for you to add the English to Hebrew function to your plugin. I appreciate your hard work, but could you please guide me and explain what steps I need to take? If possible, could you provide me with some examples? Thank you very much for your prompt responses and hard work.

**Reply by xxyzz** [reply]
I'm using the extracted Wiktionary data files downloaded from kaikki.org and dbnary to create word wise file. I don't expect the same quality as these projects but it'd be great if the data contain at least lemma, inflection forms, gloss, POS. Example sentence and IPA are not required but good to have.

You could find the data format on kaikki.org and dbnary website, for example: https://kaikki.org/dictionary/All%20languages%20combined/meaning/d/di/dictionary.html

the raw JSON wiktextract data at the bottom is used to create word wise database. But you could use other data format as long as it contains those required data.

**Reply by maorisraelii** [reply]
I hope i did it right
[he_part_1.zip](https://github.com/xxyzz/WordDumb/files/14530784/he_part_1.zip)
[he_part_2.zip](https://github.com/xxyzz/WordDumb/files/14530785/he_part_2.zip)

**Reply by maorisraelii** [reply]
Hi bro,
did you see my files? does it help to implement the heb to eng?

**Reply by xxyzz** [reply]
The Hebrew Wiktionary doesn't seem to have English word pages... I guess these files are translated to Hebrew from the English Wiktionary file? I really don't want to use machine translated data.

I think I could use the English Wiktionary's translation data to create a English to Hebrew word wise database file.

**Reply by maorisraelii** [reply]
yes this is a machine-translated data

**Comment by maorisraelii** [comment]
hi' I saw you add Hebrew on the documentation site, really thank you for that.
I updated the plugin on calibre but still do not see the option to choose Hebrew in Wordwise, can you help me with that?
thanks!

**Reply by xxyzz** [reply]
The files added are English gloss for Hebrew books, I'll create English to Hebrew files later.

**Reply by xxyzz** [reply]
English to Hebrew is enabled in the master branch, you could download the test plugin from [GitHub Actions](https://github.com/xxyzz/WordDumb/actions/runs/8398462958).

**Reply by maorisraelii** [reply]
Wow, rly rly thank you it work great but there is a little problem, there is a numbers between each translate, thank you very much for your great work
![IMG-20240323-WA0000](https://github.com/xxyzz/WordDumb/assets/109510703/343c0ae1-217e-4595-b278-bd3671759436)
![IMG-20240323-WA0001](https://github.com/xxyzz/WordDumb/assets/109510703/e8812bf5-3461-4de1-903f-1fcd4d5ea93a)

**Reply by xxyzz** [reply]
These characters are removed in this commit: https://github.com/xxyzz/Proficiency/commit/c8852cafbded331edd3729d0e59a9941c2f63b10

You'll need to download the latest test plugin from GitHub Actions and delete the old Hebrew db file in the worddumb-lemmas/en folder(inside the calibre plugin folder).

**Reply by maorisraelii** [reply]
thanks work great!
